include/common/firmware/spi_comms.hpp

```diff
@@ -2,15 +2,34 @@
 #include <cstdint>
 #include <span>
 
+#include "platform_specific_hal_conf.h"
+
+
 namespace spi {
+struct SPI_interface {
+    /*
+      Incase there are multiple handles
```

Comment:
This isn't true anymore, right. The idea is each `Spi` object will have precisely one hardware handle. So we don't need this comment. We can also make the type of `SPI_handle` the actual handle typedef, since we're including the hal_conf.h file above.